### PR TITLE
Massive improvement to source model

### PR DESCRIPTION
**Pull request by ToonDragon**
Source model can now be tested with three basic functions:
add - which adds a new source to the database
delete - which deletes a source by index
list - which lists all the sources currently in the database

Database will also load saved sources and automatically save when the exit command is entered

Added a user menu for development purposes

You can now enter an operation mode to select which app to test:
1 - for the standard address book 4
2 - to test the new model

#68

### CODE REVIEW

**Comment by ToonDragon** [comment]
The abstraction for the individual items in the source object have been added but was not included in this latest version of the model.  As it is, the extra abstraction feels unnecessary since it is just an object with a string inside.

**Review by DoItTomorrow — APPROVED** [review]
Adjusts the database model to handle sources with some basic functionality. Somewhat unsure what SourceItem* objects achieve other than add complexity, but code is functional.

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -182,6 +184,52 @@ public void stop() {
...
+                + "   1) Normal Addressbook Operation.\n"
+                + "   2) New Model Test.\n"
+                + "   0) Exit the program.");
+
+        Scanner scanner = new Scanner(System.in);
+        String input = scanner.nextLine();
+
+        switch (input) {
+            case "1":
+                System.out.println("Starting Addressbook 4.");
```
Comment (fterh) [inline src/main/java/seedu/address/MainApp.java]: You should extract these strings out as constants and put them at the top of the file

File: src/main/java/seedu/address/MainApp.java [hunk 2]
```diff
@@ -182,6 +184,52 @@ public void stop() {
...
+                SourceDatabaseCoordinationCenter coordCenter = new SourceDatabaseCoordinationCenter();
+                System.out.println("Database initialisation complete\n");
+
+                System.out.println("Instructions for commands:\n"
+                        + "add TITLE y/TYPE d/DETAILS [t/TAG]\n"
+                        + "delete INDEX\n"
+                        + "list\n"
+                        + "exit\n");
+
+                boolean loopCondition = true;
```
Comment (fterh) [inline src/main/java/seedu/address/MainApp.java]: Maybe name it more descriptively than loopCondition?

File: src/main/java/seedu/address/MainApp.java [hunk 3]
```diff
@@ -182,6 +184,52 @@ public void stop() {
...
+                while (loopCondition == true) {
```
Comment (fterh) [inline src/main/java/seedu/address/MainApp.java]: Also the `== true` part is unnecessary

File: src/main/java/seedu/address/sourcemodel/SourceDatabase.java [hunk 4]
```diff
@@ -29,6 +33,9 @@ public void deleteSourceAtIndex (int index) {
     public void deleteAllSources () {
         database.clear();
     }
+    public Source getSourceAtIndex (int index) {
+        return database.get(index);
```
Comment (fterh) [inline src/main/java/seedu/address/sourcemodel/SourceDatabase.java:37]: You're not handling IndexOutOfBoundsException!!!

File: src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    /**
+     * Takes in a user input and returns an boolean to see if the loop should end.
+     * Carries out user operations for the database.
+     *
+     * @param order The user input.
+     * @return Boolean to show if the loop should end.
+     */
+    public boolean executeOrders(String order) {
+        String orderSplit[] = order.split(" ", 2);
```
Comment (fterh) [inline src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java]: Preferred: `String[] orderSplit`

File: src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+        // There will always be a "first" element, so no array out of bounds exception can happen here
+        String orderCommand = orderSplit[0];
+
+        switch(orderCommand) {
+            case "add":
+                // Check size of array (in case user enters "add" and nothing else)
+                if (orderSplit.length < 2) {
+                    System.out.println("Invalid command parameters.\n"
```
Comment (fterh) [inline src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java]: Refactor string constants

File: src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java (hunk 5 above)
Comment (ToonDragon) [reply src/main/java/seedu/address/sourcemodel/SourceDatabaseCoordinationCenter.java]: This has been rectified in the latest commit already.

**Pull request merged**